channeldb/migrations.go

```diff
@@ -683,3 +685,162 @@ func migrateGossipMessageStoreKeys(tx *bbolt.Tx) error {
 
 	return nil
 }
+
+// migrateOutgoingPayments moves the OutgoingPayments into a new bucket format
+// where they all reside in a top-level bucket indexed by the payment hash. In
+// this sub-bucket we store information relevant to this payment, such as the
+// payment status.
+//
+// Since the router cannot handle resumed payments that have the status
```

Comment:
So then we'll potentially risk re-using a pre-image? Or this'll still be picked up by the control tower?

Reply:
After this migration, payments that were in-flight will be possible to pay again. So we do risk paying to the preimage again, but this only for payments we would have lost the preimage for anyway (because of no persistence).

Reason we don't migrate these in-flight payments is that we don't have the necessary information (we only have the payment hash) to fully populate the database. We could special case them (`"banned-paymenthashes-bucket"`), but felt it was uncommon enough case to ignore.